walletlibrary/src/main/java/com/microsoft/walletlibrary/wrapper/ManifestResolver.kt

```diff
@@ -16,14 +17,30 @@ import com.microsoft.walletlibrary.util.VerifiedIdRequestFetchException
 internal object ManifestResolver {
 
     // Fetches the issuance request from VC SDK using the url and converts it to raw request.
-    suspend fun getIssuanceRequest(uri: String): RawManifest {
+    suspend fun getIssuanceRequest(
+        uri: String,
+        requestState: String? = null,
+        issuanceCallbackUrl: String? = null
+    ): RawManifest {
         return when (val issuanceRequestResult =
             VerifiableCredentialSdk.issuanceService.getRequest(uri)) {
             is Result.Success -> {
                 val request = issuanceRequestResult.payload
                 RawManifest(request)
             }
             is Result.Failure -> {
+                val issuanceCompletionResponse = requestState?.let {
+                    IssuanceCompletionResponse(
+                        IssuanceCompletionResponse.IssuanceCompletionCode.ISSUANCE_FAILED,
+                        it,
+                        IssuanceCompletionResponse.IssuanceCompletionErrorDetails.FETCH_CONTRACT_ERROR,
+                    )
+                }
+                if (issuanceCompletionResponse != null && issuanceCallbackUrl != null)
+                    VerifiedIdCompletionCallBack.sendIssuanceCompletionResponse(
```

Comment:
what error will be returned to the developer if the Manifest resolver fails but the callback also fails?

Reply:
VerifiedIdCompletionCallbackException is thrown to the user. yeah issuance failure will be lost. I will create two separate exceptions to indicate that. I cannot multiple exception for a single failure.